test/test_sklearn_compat.py

```diff
@@ -68,5 +81,343 @@ def test_mmc(self):
+  estimator = clone(estimator)
+  estimator.set_params(preprocessor=preprocessor)
+  assert estimator.fit(tuples, y) is estimator
+
+
+@pytest.mark.parametrize('preprocessor', [None, build_data()[0]])
+@pytest.mark.parametrize('estimator, build_dataset', list_estimators,
+                         ids=ids_estimators)
+def test_pipeline_consistency(estimator, build_dataset, preprocessor):
+  # Adapted from scikit learn
+  # check that make_pipeline(est) gives same score as est
+  (_, inputs, y, _, _, _, _, preprocessor) = build_dataset(preprocessor)
+
+  def make_random_state(estimator, in_pipeline):
+    rs = {}
+    name_estimator = estimator.__class__.__name__
+    if name_estimator[-11:] == '_Supervised':
+      name_param = 'random_state'
+      if in_pipeline:
+          name_param = name_estimator.lower() + '__' + name_param
+      rs[name_param] = check_random_state(0)
+    return rs
+
+  estimator = clone(estimator)
+  estimator.set_params(preprocessor=preprocessor)
+  pipeline = make_pipeline(estimator)
+  estimator.fit(inputs, y, **make_random_state(estimator, False))
+  pipeline.fit(inputs, y, **make_random_state(estimator, True))
+
+  if hasattr(estimator, 'score'):
+    result = estimator.score(inputs, y)
+    result_pipe = pipeline.score(inputs, y)
+    assert_allclose_dense_sparse(result, result_pipe)
+
+  if hasattr(estimator, 'predict'):
+    result = estimator.predict(inputs)
+    result_pipe = pipeline.predict(inputs)
+    assert_allclose_dense_sparse(result, result_pipe)
+
+  if issubclass(estimator.__class__, TransformerMixin):
+    if hasattr(estimator, 'transform'):
+      result = estimator.transform(inputs)
+      result_pipe = pipeline.transform(inputs)
+      assert_allclose_dense_sparse(result, result_pipe)
+
+
+@pytest.mark.parametrize('preprocessor', [None, build_data()[0]])
+@pytest.mark.parametrize('estimator, build_dataset', list_estimators,
+                         ids=ids_estimators)
+def test_dict_unchanged(estimator, build_dataset, preprocessor):
+  # Adapted from scikit-learn
+  (X, tuples, y, tuples_train, tuples_test,
+   y_train, y_test, preprocessor) = build_dataset(preprocessor)
+  estimator = clone(estimator)
+  estimator.set_params(preprocessor=preprocessor)
+  if hasattr(estimator, "num_dims"):
+    estimator.num_dims = 1
+  estimator.fit(tuples, y)
+  for method in ["predict", "decision_function", "predict_proba"]:
+    if hasattr(estimator, method):
+      dict_before = estimator.__dict__.copy()
+      getattr(estimator, method)(tuples)
+      assert estimator.__dict__ == dict_before, \
+          ("Estimator changes __dict__ during %s"
+           % method)
+  for method in ["transform"]:
+    if hasattr(estimator, method):
+      dict_before = estimator.__dict__.copy()
+      # we transform only 2D arrays (dataset of points)
+      getattr(estimator, method)(X)
+      assert estimator.__dict__ == dict_before, \
+          ("Estimator changes __dict__ during %s"
+           % method)
```

Comment:
This can be refactored as it appears above as well

Reply:
Indeed, will do